controllers/flashsystemcluster_controller.go

```diff
@@ -479,14 +494,21 @@ func (r *FlashSystemClusterReconciler) ensureExporterServiceMonitor(instance *od
 		r.Log.Error(err, "failed to get exporter servicemonitor")
 		return err
 	}
+	if reflect.DeepEqual(foundServiceMonitor.Spec, expectedServiceMonitor.Spec) {
+		r.Log.Info("existing exporter deployment is expected with no change")
+		foundServiceMonitor.SetOwnerReferences(append(foundServiceMonitor.GetOwnerReferences(), newOwnerDetails))
```

Comment:
This line has no effect, as there is no call to update the object on the cluster (`r.Client.Update`)